Framework/DataHandling/test/LoadPSIMuonBinTest.h

```diff
@@ -0,0 +1,126 @@
+  static LoadPSIMuonBinTest *createSuite() { return new LoadPSIMuonBinTest(); }
+  static void destroySuite(LoadPSIMuonBinTest *suite) { delete suite; }
+
+  void test_Init() {
+    LoadPSIMuonBin alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize())
+    TS_ASSERT(alg.isInitialized())
+
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty(
+        "Filename", getTestFilePath("deltat_tdc_dolly_1529.bin")));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("OutputWorkspace", "ws"));
+  }
+
+  void test_exec() {
+    LoadPSIMuonBin alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize());
+    TS_ASSERT(alg.isInitialized());
+
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty(
+        "Filename", getTestFilePath("deltat_tdc_dolly_1529.bin")));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("OutputWorkspace", "ws"));
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+
+    MatrixWorkspace_sptr ws;
+    TS_ASSERT_THROWS_NOTHING(
+        ws = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>("ws"));
+    TS_ASSERT(ws);
+
+    TS_ASSERT_EQUALS(ws->getTitle(), "BNFSO      - Run:1529");
+    TS_ASSERT_EQUALS(ws->getLog("Field")->value(), "0.000G");
+    TS_ASSERT_EQUALS(
+        ws->getComment(),
+        "Ba3NbFe3Si2O14, crystal                                       ");
+    TS_ASSERT_EQUALS(ws->getLog("Actual Temperature1")->value(), "4.999610");
+    TS_ASSERT_EQUALS(ws->getLog("Actual Temperature2")->value(), "5.197690");
+    TS_ASSERT_EQUALS(ws->getLog("end_time")->value(), "2011-07-04T11:56:24");
+    TS_ASSERT_EQUALS(ws->getLog("start_time")->value(), "2011-07-04T10:40:23");
+    TS_ASSERT_EQUALS(ws->getLog("Spectra0 label and scalar")->value(),
+                     "Forw - 14493858");
+    TS_ASSERT_EQUALS(ws->getLog("Spectra3 label and scalar")->value(),
+                     "Rite - 38247601");
+    TS_ASSERT_EQUALS(ws->getLog("Length of Run")->value(),
+                     "10.000000MicroSeconds");
+    TS_ASSERT_EQUALS(ws->getLog("Set Temperature")->value(), "5.000K");
+
+    TS_ASSERT_EQUALS(ws->x(0).size(), 10241);
+    TS_ASSERT_EQUALS(ws->y(0).size(), 10240);
+    TS_ASSERT_EQUALS(ws->e(0).size(), 10240);
+
+    TS_ASSERT_EQUALS(ws->x(0)[0], 0);
+    TS_ASSERT_EQUALS(ws->x(0)[10240], 10);
+    TS_ASSERT_EQUALS(ws->y(0)[0], 24);
+    TS_ASSERT_EQUALS(ws->y(0)[10239], 44);
+    TS_ASSERT_EQUALS(ws->e(0)[0], std::sqrt(ws->y(0)[0]));
+    TS_ASSERT_EQUALS(ws->e(0)[10239], std::sqrt(ws->y(0)[10239]));
+
+    TS_ASSERT_EQUALS(ws->getNumberHistograms(), 4);
+
+    TS_ASSERT_THROWS_NOTHING(AnalysisDataService::Instance().remove("ws"));
+  }
+
+  void test_fileCheck() {
+    LoadPSIMuonBin alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize());
+    TS_ASSERT(alg.isInitialized());
+
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty(
+        "Filename", getTestFilePath("pid_offset_vulcan_new.dat.bin")));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("OutputWorkspace", "ws"));
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+
```

Comment:
why does this not throw at the algorithm?

Reply:
I originally wrote the test to expect a throw and even though it does throw a runtime error the algorithm itself does not throw anything, itself I can achieve the same effect by making sure the algorithm never finishes because no workspace is in the ads.

Reply:
I think the algorithm should throw something if it fails.